### PR TITLE
Raw arrays - InventoryPanel

### PR DESCRIPTION
**Pull request by fredg1**

### CODE REVIEW

**Comment by gausie** [comment]
Why is this better?

File: src/net/sourceforge/kolmafia/swingui/panel/InventoryPanel.java [hunk 1]
```diff
@@ -86,15 +87,15 @@ public void addFilters() {
...
-    this.equipmentFilters[7] = new FilterRadioButton("familiar");
+    this.equipmentFilters = new ArrayList<>(8);
+    this.equipmentFilters.add(new FilterRadioButton("weapons", true));
+    this.equipmentFilters.add(new FilterRadioButton("offhand"));
+    this.equipmentFilters.add(new FilterRadioButton("hats"));
+    this.equipmentFilters.add(new FilterRadioButton("back"));
+    this.equipmentFilters.add(new FilterRadioButton("shirts"));
+    this.equipmentFilters.add(new FilterRadioButton("pants"));
+    this.equipmentFilters.add(new FilterRadioButton("accessories"));
+    this.equipmentFilters.add(new FilterRadioButton("familiar"));
```
Comment (gausie) [inline src/net/sourceforge/kolmafia/swingui/panel/InventoryPanel.java:98]:
```suggestion
    this.equipmentFilters = Arrays.asList(
        new FilterRadioButton("weapons", true),
        new FilterRadioButton("offhand"),
        new FilterRadioButton("hats"),
        new FilterRadioButton("back"),
        new FilterRadioButton("shirts"),
        new FilterRadioButton("pants"),
        new FilterRadioButton("accessories"),
        new FilterRadioButton("familiar"),
    );
```

Comment (gausie) [reply src/net/sourceforge/kolmafia/swingui/panel/InventoryPanel.java:98]: You'll need to fix imports for this obviously.

**Comment by fredg1** [comment]
> Why is this better?

https://github.com/kolmafia/kolmafia/pull/148#issuecomment-949555223

**Comment by gausie** [comment]
> > Why is this better?
> 
> [#148 (comment)](https://github.com/kolmafia/kolmafia/pull/148#issuecomment-949555223)

But how does that apply here?

**Comment by fredg1** [comment]
> > > Why is this better?
> > 
> > 
> > [#148 (comment)](https://github.com/kolmafia/kolmafia/pull/148#issuecomment-949555223)
> 
> But how does that apply here?

Because `FilterRadioButton` *is a child class of* a parameterized class.

```java
public class InventoryPanel<E> extends ItemTableManagePanel<E> {
  private class FilterRadioButton extends JRadioButton { [...] }
}
```

Compiler wants/needs to know what is the type of its parent (e.g. `InventoryPanel<String>.FilterRadioButton`)

**Comment by gausie** [comment]
ah its a child class, i understand. cool thanks!

**Review by gausie — APPROVED** [review]

**Pull request merged**